### PR TITLE
Custom DB Docs

### PR DESCRIPTION
**Pull request by khorne3**
https://auth0-docs-content-pr-6406.herokuapp.com/docs/connections/database/custom-db

### CODE REVIEW

File: articles/connections/database/custom-db/index.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+* Create a [create and configure a custom database connection](/connections/database/custom-db/create-db-connection) using the [Auth0 dashboard](${manage_url}).
+* Make sure that your database has the appropriate fields to store user profiles attributes, such as **id**, **nickname**, **email**, and **password**. See [Auth0 Normalized User Profile](/user-profile/normalized) for details on Auth0's user profile schema and the expected fields.
+* Provide database action scripts to configure the database for use as an identity provider.
+
+## Before you begin
+
+Here are some things to know before you begin the process of setting up your database for use as an identity provider.
+
+* You'll write your database action scripts in JavaScript. The scripts run in a [Webtask](https://webtask.io/) environment that supports JavaScript and [select Node.js libraries](https://tehsis.github.io/webtaskio-canirequire/).
```
Comment (mpaktiti) [inline articles/connections/database/custom-db/index.md]: The `canirequire` thingy was moved here https://auth0-extensions.github.io/canirequire/
There is a redirect in place from tehsis repo, but we might as well set the correct URL.

Also, marketing-wise I'm not sure if we still want to refer to webtasks as webtasks and not Extend. I think gblock mentioned sth to me some time ago, can you check with him?

File: articles/connections/database/custom-db/index.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  * MySQL
+  * Oracle
+  * PostgreSQL
+  * SQLServer
+  * Windows Azure SQL Database
+  * Web services accessed via Basic Auth
+
+  Essentially, you can connect to any kind of database or web service with a properly-configured custom script.
+
+* [Update Users Using Your Database](/user-profile/customdb) has information on updating user profile fields.
```
Comment (mpaktiti) [inline articles/connections/database/custom-db/index.md]: I know you have put some links inline, but I did expect to see a what's next section at the end listing the rest of the docs

File: articles/connections/database/custom-db/create-db-connection.md [hunk 3]
```diff
@@ -0,0 +1,152 @@
+---
+description: Learn how to create a database connection.
+toc: true
+topics:
+    - connections
+    - custom-database
+contentType: how-to
+useCase: customize-connections
+---
+Create an Auth0 database connection
```
Comment (mpaktiti) [inline articles/connections/database/custom-db/create-db-connection.md]: Missing header, check https://auth0-docs-content-pr-6406.herokuapp.com/docs/connections/database/custom-db/create-db-connection

File: articles/connections/database/custom-db/create-db-connection.md [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+
+The first thing you will do is create a database connection in Auth0:
```
Comment (mpaktiti) [inline articles/connections/database/custom-db/create-db-connection.md]: Feels like we should have an intro here explaining what the doc is about. If someone lands her directly, and not through the index, they see this and might wonder what's this first thing is about.

File: articles/connections/database/custom-db/create-db-connection.md [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+* Applications
+* Try Connection
+
+4. Switch over to the **Custom Database** tab.
+
+5. Toggle the **Use my own database** switch to enable this feature.
+
+![Custom database tab](/media/articles/connections/database/custom-database.png)
+
+## Step 2: Create database action scripts
```
Comment (mpaktiti) [inline articles/connections/database/custom-db/create-db-connection.md:48]: Looks like there should be a header before that for `Step 1`, we can't start from 2

File: articles/connections/database/custom-db/create-db-connection.md [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+}
+```
+
+## Summary
+
+In this article, we showed you how to configure your database for use with Auth0 as an identity provider. You: 
+
+1. Created an Auth0 database connection
+2. Created database action scripts
+3. Added configuration parameters
```
Comment (mpaktiti) [inline articles/connections/database/custom-db/create-db-connection.md]: what's the next logical step? do we want a next-steps section here or are we removing this notion?

File: articles/connections/database/custom-db/error-handling.md [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+topics:
+    - connections
+    - custom-database
+contentType: how-to
+useCase:
+    - customize-connections
+---
+# Custom Database Error Handling
+
+In this article, we will show you how you can handle errors resulting from your custom database connection.
```
Comment (mpaktiti) [inline articles/connections/database/custom-db/error-handling.md]: This article does not really say how to handle errors, all I can see is the different types of errors and how I can return one.

File: articles/connections/database/custom-db/lock.md [hunk 8]
```diff
@@ -0,0 +1,11 @@
+---
+description: Learn how to use Lock with your custom database.
+crews: crew-2
+---
+# Using Lock with Custom Databases
```
Comment (mpaktiti) [inline articles/connections/database/custom-db/lock.md]: I'm wondering whether this is  doc that we should keep. It doesn't say much so my 2c is that we should either remove it or enhance it.

File: articles/connections/database/custom-db/lock.md [hunk 9]
```diff
@@ -0,0 +1,11 @@
+---
+description: Learn how to use Lock with your custom database.
+crews: crew-2
```
Comment (mpaktiti) [inline articles/connections/database/custom-db/lock.md]: these crew-2 metadata were never used so we might as well remove them (there is not even a crew-2 in the company any longer). no need to do this as part of this PR but please add this in trello

File: articles/connections/database/custom-db/lock.md [hunk 10]
```diff
@@ -0,0 +1,11 @@
...
+---
+# Using Lock with Custom Databases
+
+If you use [Lock](libraries#lock), enabling the database connection lets users enter their username and password on the Auth0 Login widget. Once entered, this data is passed to your scripts.
+
+![Auth0 login widget](/media/articles/connections/database/mysql/db-connection-widget.png)
+
+<%= include('../../_quickstart-links.md') %>
```
Comment (mpaktiti) [inline articles/connections/database/custom-db/lock.md]: I don't see the logical connection between this doc and the next steps listed here

File: articles/connections/database/custom-db/metadata.md [hunk 11]
```diff
@@ -0,0 +1,9 @@
+---
+description: Learn how to work with metadata when using your database as an identity provider.
+crews: crew-2
+---
+# Metadata and Custom Databases
+
```
Comment (mpaktiti) [inline articles/connections/database/custom-db/metadata.md]: what's this doc about? in which case should I read it?

how are the metadata relevant to the IP whitelisting that follows?

File: articles/connections/database/custom-db/templates/change-password.md [hunk 12]
```diff
@@ -0,0 +1,534 @@
...
+description: Custom DB script templates for changing a user's password
+toc: true
+topics:
+    - connections
+    - custom-database
+contentType: how-to
+useCase:
+    - customize-connections
+---
+# Custom Database Script Templates: Change Password
```
Comment (mpaktiti) [inline articles/connections/database/custom-db/templates/change-password.md:11]: this doc means that we should update it every time some script gets updated. are we sure we want to do that? did someone ask for it specifically? wouldn't it make more sense to have a revert button in the dashboard?

File: articles/connections/database/custom-db/templates/create.md [hunk 13]
```diff
@@ -0,0 +1,609 @@
...
+toc: true
+topics:
+    - connections
+    - custom-database
+contentType: how-to
+useCase:
+    - customize-connections
+---
+# Custom Database Script Templates: Create Users
+
```
Comment (mpaktiti) [inline articles/connections/database/custom-db/templates/create.md:12]: same comment with `/custom-db/templates/change-password.md`

File: articles/connections/database/custom-db/templates/delete.md [hunk 14]
```diff
@@ -0,0 +1,362 @@
...
+description: Custom DB script templates for user deletion
+toc: true
+topics:
+    - connections
+    - custom-database
+contentType: how-to
+useCase:
+    - customize-connections
+---
+# Custom Database Script Templates: Delete Users
```
Comment (mpaktiti) [inline articles/connections/database/custom-db/templates/delete.md:11]: same comment with `/custom-db/templates/change-password.md`

File: articles/connections/database/custom-db/templates/get-user.md [hunk 15]
```diff
@@ -0,0 +1,287 @@
...
+description: Custom DB script templates for user search
+toc: true
+topics:
+    - connections
+    - custom-database
+contentType: how-to
+useCase:
+    - customize-connections
+---
+# Custom Database Script Templates: Get Users
```
Comment (mpaktiti) [inline articles/connections/database/custom-db/templates/get-user.md:11]: same comment with `/custom-db/templates/change-password.md`

File: articles/connections/database/custom-db/templates/index.md [hunk 16]
```diff
@@ -0,0 +1,23 @@
...
+description: Custom DB script templates
+topics:
+    - connections
+    - custom-database
+contentType: index
+useCase:
+    - customize-connections
+---
+# Custom Database Script Templates
+
```
Comment (mpaktiti) [inline articles/connections/database/custom-db/templates/index.md:11]: same comment with `/custom-db/templates/change-password.md`

File: articles/connections/database/custom-db/templates/index.md [hunk 17]
```diff
@@ -0,0 +1,23 @@
...
+Auth0 provides the following custom database change password script templates that you can use when implementing password changing functionality for your users.
+
+While Auth0 has populated default templates in the Dashboard script editor, you can use the following links to recover the original code and notes once you've made and saved edits.
+
+::: next-steps
```
Comment (mpaktiti) [inline articles/connections/database/custom-db/templates/index.md]: When there are over 4-5 items this element looks bad I think. Consider replacing it with bullets https://auth0-docs-content-pr-6406.herokuapp.com/docs/connections/database/custom-db/templates

File: articles/connections/database/custom-db/templates/login.md [hunk 18]
```diff
@@ -0,0 +1,731 @@
...
+description: Custom DB script templates for user login
+toc: true
+topics:
+    - connections
+    - custom-database
+contentType: how-to
+useCase:
+    - customize-connections
+---
+# Custom Database Script Templates: Login
```
Comment (mpaktiti) [inline articles/connections/database/custom-db/templates/login.md:11]: same comment with `/custom-db/templates/change-password.md`

File: articles/connections/database/custom-db/templates/verify.md [hunk 19]
```diff
@@ -0,0 +1,461 @@
...
+description: Custom DB script templates for user verification
+toc: true
+topics:
+    - connections
+    - custom-database
+contentType: how-to
+useCase:
+    - customize-connections
+---
+# Custom Database Script Templates: Verify Users
```
Comment (mpaktiti) [inline articles/connections/database/custom-db/templates/verify.md:11]: same comment with `/custom-db/templates/change-password.md`

File: articles/connections/database/custom-db/troubleshoot.md [hunk 20]
```diff
@@ -0,0 +1,15 @@
+---
+description: Learn how to troubleshoot errors when using your database as an identity provider.
+crews: crew-2
+---
+# Troubleshoot Errors When Using Custom Databases
+
```
Comment (mpaktiti) [inline articles/connections/database/custom-db/troubleshoot.md]: this is a bit dry, it's missing an intro/scope of the doc

**Review by mpaktiti — CHANGES_REQUESTED** [review]
Hey Katie, left some comments, let me know if you wanna chat

File: articles/connections/database/custom-db/templates/change-password.md (hunk 12 above)
Comment (khorne3) [reply articles/connections/database/custom-db/templates/change-password.md:11]: The answer to this is yes, there are issues with people losing out on the notes and things we populate in the dashboard. I agree w/ your revert suggestion, but what happens if the user wants to keep their script but also refer to the use notes we include in the text editor? For the time being, I understand we're creating a maintenance issue, but there are crucial items we don't want the customers to miss.

**Comment by khorne3** [comment]
Currently, the only outstanding item is whether we want to use Webtask or Extend. I can update when Glenn returns OOF.

**Review by mpaktiti — APPROVED** [review]

**Pull request merged**